src/RecoveryServices/RecoveryServices.Backup.Autorest/custom/Cmdlets/Policy/Edit-AzRecoveryServicesBackupSchedulePolicyClientObject.ps1

```diff
@@ -0,0 +1,408 @@
+﻿function Edit-AzRecoveryServicesBackupSchedulePolicyClientObject {
+	[OutputType('Microsoft.Azure.PowerShell.Cmdlets.RecoveryServices.Models.Api20230201.IProtectionPolicy')]
+    [CmdletBinding(PositionalBinding=$false)]
+    # RsvRef: should we call it workload type
+    [Microsoft.Azure.PowerShell.Cmdlets.RecoveryServices.Description('Edits the schedule policy in the specified backup policy object.')]
+
+	param (
+        [Parameter(ParameterSetName='ModifySchedulePolicy', Mandatory, HelpMessage='Specifies the policy to be edited.')]
+        [Microsoft.Azure.PowerShell.Cmdlets.RecoveryServices.Models.Api20230201.IProtectionPolicy]
+        ${Policy},
+      
+        # AzureVM specific parameters
+
+        [Parameter(ParameterSetName='ModifySchedulePolicy', HelpMessage='Specifies the policy sub type for AzureVM.')]
+        [ValidateSet("Standard", "Enhanced", ErrorMessage = "Invalid value for PolicySubType. Please provide a valid policy sub type. Valid values are 'Standard' and 'Enhanced'.")]
+        [string]
+        ${PolicySubType} = "Standard",
+
+        [Parameter(ParameterSetName='ModifySchedulePolicy', HelpMessage='Specifies the interval between backups in hours.')]
+        [ValidateSet(4, 6, 8, 12, ErrorMessage = "Invalid value for HourlyInterval. Please provide a valid interval. Valid values are 4, 6, 8, and 12.")]
+        [AllowNull()]
+        [System.Nullable[int]]
+        ${HourlyInterval} = $null,
+
+        [Parameter(ParameterSetName='ModifySchedulePolicy', HelpMessage='Specifies the duration over which backup is taken in hours.')]
+        [ValidateSet(4, 8, 12, 16, 20, 24, ErrorMessage = "Invalid value for HourlyScheduleWindowDuration. Please provide a valid duration. Valid values are 4, 8, 12, 16, 20, and 24.")]
+        [AllowNull()]
+        [System.Nullable[int]]
+        ${HourlyScheduleWindowDuration} = $null,
+
+        # SAPHANA specific parameters
+
+        [Parameter(ParameterSetName='ModifySchedulePolicy', HelpMessage='Specifies whether the user needs to make a log backup')]
+        [AllowNull()]
+        [Nullable[Boolean]]
+        ${EnableLogBackup},
+
+        [Parameter(ParameterSetName='ModifySchedulePolicy', HelpMessage='Specifies the frequency of log backups in minutes')]
+        [ValidateSet(15, 30, 60, 120, 240, 480, 720, 1440, ErrorMessage = "Invalid value for LogBackupFrequency. Please provide a valid frequency. Valid values are 15, 30, 60, 120, 240, 480, 720, and 1440.")]
+        [AllowNull()]
+        [System.Nullable[int]]
+        ${LogBackupFrequency} = $null,
+
+        [Parameter(ParameterSetName='ModifySchedulePolicy', HelpMessage='Specifies whether the user wants to enable differential backup.')]
+        [AllowNull()]
+        [Nullable[Boolean]]
+        ${EnableDifferentialBackup} = 0,
+
+        [Parameter(ParameterSetName='ModifySchedulePolicy', HelpMessage='Specifies the days of the week for differential backup.')]
+        [ValidateScript({
```

Comment:
@t-anssingh102 : pls take a look

Reply:
@vidai-msft Sir actually `ValidateSet` will not serve our purpose here I think because `ScheduleRunDay`, `DifferentialRunDay` etc can be any subset of the set of days of the week. And if we want to check this through `ValidateSet` we'll have to list all possible subsets in the set i.e. 2^7 = 128 which is not feasible I think.

Reply:
oh, yes. The type is string array. ValidateScript sould be more convenient. However, please kindly note a side effect of the `$_ -ne $null` as empty array would be considered as $false when compared with $null using `@() -ne $null` while `$null -ne @()` is $true.

Reply:
Alright I'll fix this